packages/presence-signal/README.md

```diff
@@ -0,0 +1,173 @@
+    transitionDuration: 500,
+  });
+
+  return (
+    <div
+      style={{
+        padding: "2em",
+        margin: "2em",
+        "border-radius": "2em",
+        "box-shadow": "-5px 0px 10px rgba(0, 0, 0, 0.2)",
+      }}
+    >
+      <button onclick={() => setShowStuff(!showStuff())}>{`${
+        showStuff() ? "Hide" : "Show"
+      } stuff`}</button>
+      <Show when={isMounted()}>
+        <div
+          style={{
+            transition: "all .5s ease",
+            opacity: isVisible() ? "1" : "0",
+            transform: isVisible() ? "translateX(0)" : "translateX(50px)",
+          }}
+        >
+          I am the stuff!
+        </div>
+      </Show>
+    </div>
+  );
+};
+```
+
+## API
+
+```tsx
+type Options = {
+  /** Duration in milliseconds used both for enter and exit transitions. */
+  transitionDuration: number;
+  /** Duration in milliseconds used for enter transitions (overrides `transitionDuration` if provided). */
+  enterTransitionDuration: number;
+  /** Duration in milliseconds used for exit transitions (overrides `transitionDuration` if provided). */
+  exitTransitionDuration: number;
+  /** Opt-in to animating the entering of an element if `isVisible` is `true` during the initial mount. */
+  initialEnter?: boolean;
+};
+createPresenceSignal(
+  /** Indicates whether the component that the resulting values will be used upon should be visible to the user. */
+  isVisible: Accessor<boolean>,
+  /** Options may be a signal getter if you need to update it */
+  opts: Options | Accessor<Options>
+): {
+  /** Should the component be returned from render? */
+  isMounted: Accessor<boolean>;
+  /** Should the component have its visible styles applied? */
+  isVisible: Accessor<boolean>;
+  /** Is the component either entering or exiting currently? */
+  isAnimating: Accessor<boolean>;
+  /** Is the component entering currently? */
+  isEntering: Accessor<boolean>;
+  /** Is the component exiting currently? */
+  isExiting: Accessor<boolean>;
+}
+```
+
+## `createPresenceSwitchSignal`
+
+If you have multiple items where only one is visible at a time, you can use the supplemental `createPresenceSwitchSignal` utility to animate the items in and out. Previous items will exit before the next item transitions in.
+
+### API
+
+```tsx
+createPresenceSwitchSignal<ItemType>(
+  /** The current item that should be visible. If `undefined` is passed, the previous item will animate out. */
+  item: Accessor(ItemType | undefined>,
+  /** The same `opts` argument of `createPresenceSignal`. */
+  opts: Options | Accessor<Options>
+): {
+  /** The item that should currently be rendered. */
+  mountedItem: Accessor<ItemType | undefined>;
+  /** Returns all other properties from `createPresenceSignal`. */
+  ...rest
+}
+```
+
+### Example
+
+```jsx
+const SecondExample = () => {
+  const items = ["foo", "bar", "baz", "qux"];
+  const [item, setItem] =
+    (createSignal < typeof items[number]) | (undefined > items[0]);
```

Comment:
Did something happen to the formatting here? it doesn't seem real

Reply:
Good catch 😅  I think it was an autoformatting issue. This block was previously marked as `jsx` and not `tsx` but this line was supposed to be

```tsx
const [item, setItem] = createSignal<(typeof items)[number] | undefined>(items[0]);
```

Fixed 👍